### PR TITLE
Add integration tests for RC manage version operations

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add integration tests for `listVersions`, `getTemplateAtVersion`, and `rollback` operations
- Add integration tests for `versions` metadata in Remote Config templates
- Remove undefined values in list versions options, and add unit tests
- Check for empty versions lists in `listVersions`, and add unit tests

Note: Added `release:stage` to trigger integration tests. Merging to ~`remote-config-vc`~ `master`.

### CODE REVIEW

**Comment by lahirumaramba** [comment]
Note: That failing integration test is from ML. Fix is in `master`

File: src/remote-config/remote-config.ts [hunk 1]
```diff
@@ -144,8 +144,10 @@ export class RemoteConfig implements FirebaseServiceInterface {
   public listVersions(options?: ListVersionsOptions): Promise<ListVersionsResult> {
     return this.client.listVersions(options)
       .then((listVersionsResponse) => {
+        const versions = (typeof listVersionsResponse.versions == 'undefined') ? [] :
```
Comment (hiranya911) [inline src/remote-config/remote-config.ts:147]:
```
const versions = listVersionsResponse.versions?.map(...) ?? [];
```

File: test/integration/remote-config.spec.ts [hunk 2]
```diff
@@ -57,16 +57,22 @@ const VALID_PARAMETER_GROUPS = {
...
+    tagColor: 'INDIGO',
+  },
+  {
+    name: 'android',
+    expression: 'device.os == \'android\'',
+    tagColor: 'GREEN',
+  },
+];
+
+const VALID_VERSION: admin.remoteConfig.Version = {
```
Comment (hiranya911) [inline test/integration/remote-config.spec.ts:73]: Just wondering. Are these explicit type annotations really required?

File: src/remote-config/remote-config-api-client.ts [hunk 3]
```diff
@@ -237,6 +237,9 @@ export class RemoteConfigApiClient {
   public listVersions(options?: ListVersionsOptions): Promise<ListVersionsResult> {
     if (typeof options !== 'undefined') {
       options = this.validateListVersionsOptions(options);
+      // Remove undefined fields from options
+      Object.keys(options).forEach(key =>
+        (options as any)[key] === undefined && delete (options as any)[key])
```
Comment (hiranya911) [inline src/remote-config/remote-config-api-client.ts:242]: I feel like this logic belongs in the `validateListVersionsOptions` method.

Also `typeof ... === 'undefined'`

File: test/integration/remote-config.spec.ts [hunk 4]
```diff
@@ -143,6 +157,87 @@ describe('admin.remoteConfig', () => {
...
+      versionOneNumber = activeTemplate.version!.versionNumber!;
+
+      // publish another template to create a second version number
+      activeTemplate.version = { description: versionTwoDescription };
+      activeTemplate = await admin.remoteConfig().publishTemplate(activeTemplate)
+      expect(activeTemplate.version).to.be.not.undefined;
+      versionTwoNumber = activeTemplate.version!.versionNumber!;
+    });
+
+    it('verfy that getTemplateAtVersion() returns the requested template version v1', () => {
```
Comment (hiranya911) [inline test/integration/remote-config.spec.ts:189]: Use the familiar wording style. `should return....`

File: test/integration/remote-config.spec.ts [hunk 5]
```diff
@@ -143,6 +157,87 @@ describe('admin.remoteConfig', () => {
...
+      return admin.remoteConfig().getTemplateAtVersion(versionOneNumber)
+        .then((template) => {
+          expect(template.etag).matches(/^etag-[0-9]*-[0-9]*$/);
+          expect(template.version).to.be.not.undefined;
+          expect(template.version!.versionNumber).equals(versionOneNumber);
+          expect(template.version!.description).equals(versionOneDescription);
+        });
+    });
+
+    it('verfy that getTemplateAtVersion() returns the requested template version v2', () => {
```
Comment (hiranya911) [inline test/integration/remote-config.spec.ts:199]: Seems redundant.

File: src/remote-config/remote-config-api-client.ts (hunk 3 above)
Comment (lahirumaramba) [reply src/remote-config/remote-config-api-client.ts]: Good call. Moved to `validateListVersionsOptions`

File: src/remote-config/remote-config.ts (hunk 1 above)
Comment (lahirumaramba) [reply src/remote-config/remote-config.ts]: Thanks!

File: test/integration/remote-config.spec.ts (hunk 2 above)
Comment (lahirumaramba) [reply test/integration/remote-config.spec.ts]: I removed the type annotations for `VALID_VERSION`. It looks like we need the type annotation for `VALID_CONDITIONS` unless I use a `const INDIGO: admin.remoteConfig.TagColor = 'INDIGO'` for tag color.

File: test/integration/remote-config.spec.ts (hunk 5 above)
Comment (lahirumaramba) [reply test/integration/remote-config.spec.ts]: This was to confirm that we are not getting the most recent template and having two versions confirms it. I see your point though :)
Will remove the `expect`s here, but leave the templates for `listVersions` tests.

**Pull request merged**